### PR TITLE
feat: can have space in project name

### LINKED ISSUE #761: Allow spaces in Project name
The `project_name` field in `data_description` doesn't allow spaces. We want this to be able to use the project names from the funding smartsheet which do have spaces. Can we change the regex to allow spaces?

### PR DESCRIPTION
**Pull request by micahwoodard**
fixes #761 
- allow project name to have spaces

### CODE REVIEW

**Comment by micahwoodard** [comment]
I changed the NO_SPECIAL_CHARS to allow spaces but can create new variable if that is what's needed

**Review by saskiad — APPROVED** [review]

**Comment by micahwoodard** [comment]
Changed to have an additional variable in DataRegex to allow no special characters but space

File: src/aind_data_schema/core/data_description.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class DataRegex(str, Enum):
     )
     NO_UNDERSCORES = "^[^_]+$"
     NO_SPECIAL_CHARS = '^[^<>:;"/|? \\_]+$'
+    NO_SPECIAL_CHARS_EXCEPT_SPACE = '^[^<>:;"/|? \\_]+$'
```
Comment (jtyoung84) [inline src/aind_data_schema/core/data_description.py:41]: This is the same regex pattern as the line above

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
Can you add a couple of unit tests that test the regex patterns are doing what's expected? You don't need to create entire models; you can pull the regex patterns by `DataDescription.NO_SPECIAL_CHARS`, etc.

**Comment by micahwoodard** [comment]
Added some test for the different patterns but may need to add or change to better test

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**